suro-core/src/main/java/com/netflix/suro/sink/ThreadPoolQueuedSink.java

```diff
@@ -40,7 +40,7 @@ public boolean offer(Runnable runnable) {
                 maxPoolSize == 0 ? 10 : maxPoolSize,
                 10, TimeUnit.SECONDS,
                 jobQueue,
-                new ThreadFactoryBuilder().setNameFormat(threadFactoryName + "-Sender-%d").build());
+                new ThreadFactoryBuilder().setNameFormat(threadFactoryName + "-Sender-%d").setDaemon(true).build());
```

Comment:
I don't believe make it non-daemon will avoid msg loss though. it will only prevent JVM from shutdown.

this method shutdown the thread pool.
@Override
    protected void innerClose() {
        senders.shutdown();
        try {
            senders.awaitTermination(jobTimeout == 0 ? Long.MAX_VALUE : jobTimeout, TimeUnit.MILLISECONDS);
            if (!senders.isTerminated()) {
                log.error(this.getClass().getSimpleName() + " not terminated gracefully");
            }
        } catch (InterruptedException e) {
            log.error("Interrupted: " + e.getMessage());
        }
    }